os/dnd.h

```diff
@@ -0,0 +1,125 @@
+// LAF OS Library
+// Copyright (C) 2024  Igara Studio S.A.
+//
+// This file is released under the terms of the MIT license.
+// Read LICENSE.txt for more information.
+
+#ifndef OS_DND_H_INCLUDED
+#define OS_DND_H_INCLUDED
+#pragma once
+
+#include "base/paths.h"
+#include "base/debug.h"
+#include "gfx/point.h"
+/*#include "os/surface.h"*/
+
+#include <algorithm>
+#include <cstddef>
+#include <functional>
+#include <memory>
+#include <utility>
+#include <variant>
+/*#include <vector>*/
+
+namespace os {
+
+  class Window;
+
+  // Operations that can be supported by source and target windows in a drag
+  // and drop operation.
+  enum DropOperation {
+    None = 0,
+    Copy = 1,
+    Move = 2,
+    Link = 4,
+    Any = Copy | Move | Link,
+  };
+
+  // Types of representations supported for each DragDataItem.
+  enum DragDataItemType {
+    Paths,
+    /*Images*/
+  };
+
+  using DragDataItemSupportedType = std::variant<base::paths>;
+
+  // Interface to get dragged data from the platform's implementation.
+  class DragDataProvider {
+  public:
+    virtual ~DragDataProvider() {}
+    virtual base::paths getPaths() = 0;
+    /*virtual std::vector<Surface*> getImages() = 0;*/
+  };
+
+  class DragEvent {
+  public:
+    DragEvent(os::Window* target,
+              DropOperation supportedOperations,
+              const gfx::Point& dragPosition,
+              std::unique_ptr<DragDataProvider> dataProvider)
+      : m_target(target)
+      , m_supportedOperations(supportedOperations)
+      , m_position(dragPosition)
+      , m_dataProvider(std::move(dataProvider)) {}
+
+    // Destination window of the DragEvent.
+    os::Window* target() const { return m_target; }
+    DropOperation dropResult() const { return m_dropResult; }
+    DropOperation supportedOperations() const { return m_supportedOperations; }
+    bool acceptDrop() const { return m_acceptDrop; }
+    const gfx::Point& position() { return m_position; }
+
+    template<DragDataItemType IT,
+             typename T = std::variant_alternative_t<int(IT), DragDataItemSupportedType>>
+    constexpr T getItems() {
+      if constexpr (IT == Paths) {
+        return m_dataProvider->getPaths();
+      }
+      /*
+      else if constexpr (IT == Images) {
+        return m_dataProvider->getImages();
+      }
+      */
+      return T();
+    }
```

Comment:
I'd remove this and just use the DragDataProvider directly in the client side (we need a getter for the m_dataProvider). And the DragDataProvider a way to check the type of data/or included types in the data (?)